apps/src/dance/generators.ts

```diff
@@ -0,0 +1,27 @@
+import {Block} from 'blockly';
+import {DANCE_AI_FIELD_NAME} from './ai/constants';
+
+type Generator = (block: Block) => string | string[];
+
+const generators: {[blockName: string]: Generator} = {
+  ai: (block: Block) => {
+    return `
+      // Custom block generators allow us to perform some other complex logic
+
+      // const value = API.getSomeValue(block.getFieldValue('some_field'));
+      // API.setSomeValue(value);
+      // API.prepareForAi(block.id);
+
+      ai(${block.getFieldValue(DANCE_AI_FIELD_NAME)});
+      
+      // API.endAi();
+      `;
+  },
+  setBackgroundEffectWithPaletteAI: (block: Block) => {
+    const effect = block.getFieldValue('EFFECT');
+    const palette = block.getFieldValue('PALETTE');
+    return `setBackgroundEffectWithPalette('${effect}', '${palette}');`;
```

Comment:
Some small but nice benefits from using this system for this block in particular
- this block is the same as the `setBackgroundEffectWithPalette` block, just with more dropdown options. However in the existing system, since it's a separate block with a separate name, we also had to add a new function in the dance API layer corresponding to the new name that did the exact same thing. Instead here, we can internally call into the existing `setBackgroundEffectWithPalette` (non-AI) function for this block.
- We can manually surround the block effect and palette values with quotes so that we don't need to make sure to include quotes around block values in the config json.